apps/emqx_gateway_coap/src/emqx_coap_channel.erl

```diff
@@ -204,9 +204,9 @@ handle_deliver(
 
 handle_timeout(_, {keepalive, NewVal}, #channel{keepalive = KeepAlive} = Channel) ->
     case emqx_keepalive:check(NewVal, KeepAlive) of
-        {ok, NewKeepAlive} ->
+        {ok, _} ->
             Channel2 = ensure_keepalive_timer(fun make_timer/4, Channel),
-            {ok, Channel2#channel{keepalive = NewKeepAlive}};
```

Comment:
I think this still needs to replace the old keepalive, as `check` returns `#keepalive{}` updated with `NewVal`.  Otherwise, it will always "succeed" (always thinks it has received timely heartbeats).  :thinking: